BLUESPAWN-client/src/hunt/hunts/HuntT1035.cpp

```diff
@@ -46,35 +46,36 @@ namespace Hunts {
 
 			detections += 2;
 		}
-
+		
 		auto name = key.GetName();
 		name = name.substr(name.find_last_of(L"\\") + 1);
-
 		RegistryKey subkey = RegistryKey{ HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Services\\" + name + L"\\Parameters" };
 
-		if (subkey.Exists() && subkey.ValueExists(L"ServiceDll")) {
-			auto filepath2 = GetImagePathFromCommand(subkey.GetValue<std::wstring>(L"ServiceDll").value());
-
-			if (!FileSystem::CheckFileExists(filepath2)) {
-				return detections;
+		for (auto regkey : { key, subkey }) {
+			if (!regkey.Exists()) {
+				continue;
 			}
 
-			FileSystem::File servicedll = FileSystem::File(filepath2);
+			std::wstring value = L"ServiceDll";
+			if (regkey.ValueExists(value)) {
+				auto filepath2 = GetImagePathFromCommand(regkey.GetValue<std::wstring>(value).value());

```

Comment:
FileSystem::SearchPathExecutable is the function you want. Dlls are not commands. The functions behave differently, and this may result in incorrect DLLs being checked